### ISSUE TITLE
Evaluate feasibility of ZKP over ECDSA-based signed message

### ISSUE THREAD

**Issue by dwaite** [ietf-wg-jose/json-web-proof#148]
A recent presentation from @abhvious showed a ZKP over a mdoc presentation. this used specifically optimized processing and circuit design to achieve acceptable performance, while also allowing for use of P256 and SHA-256 to meet the reality of today's secure enclaves/elements.

It would be useful to evaluate these techniques once they are published to see how they apply to JWP, and if there are changes we could make within JWP to optimize ZKP evaluation.

**Comment by selfissued** [comment]
I agree.  @abhvious, is there a public description of the approach at this point?

**Comment by selfissued** [comment]
Once a public description of this algorithm is published, I agree that it would make sense to investigate it - probably in a separate algorithm spec.

**Comment by selfissued** [comment]
https://datatracker.ietf.org/doc/draft-google-cfrg-libzk/

**Comment by abhvious** [comment]
We also published an academic paper: https://eprint.iacr.org/2024/2010 that describes the rationale behind the system.

I would like to learn more about the JWP usecase you mention.
